### PR TITLE
1154/classnotes

### PR DESCRIPTION
**Pull request by iamsitting**
See #1154 
- [x]  Assign class notes by week
  - Drops assignment from 9 minutes to ~45 sec
- [x] View report by week (only after after assigning by week)
  - After assigning takes them to a list of assignments for that week

### CODE REVIEW

**Comment by JoannaRumbley** [comment]
I talked to brother Andrew and he gave a lot of feedback. I feel like it would be helpful for us to discuss in person too, but here's the summarized issues:

- [ ] If trainee submits paper copy, TAs have no way to approve it

- [ ] Cannot see who still has not completed their class notes (old server also had a toggle for "Received" and "Not received")

- [ ] Cannot ensure class notes were assigned correctly (does server account for cases such as 3 absences that were covered by a service leave slip? and other cases outlined in the info manual?)

- [ ] Need a page like the "Class Notes Detail" page on the old server

**Comment by iamsitting** [comment]
1. Will work on this.
2. Will work on this.
3. It should.
4. Is the "Class Notes Detail" page just a summary of class notes for a single trainee? It's hard to tell because there's no data on the old server?

**Comment by iamsitting** [comment]
I think I understand now. It's a summary for trainees, but TAs can use it to view a summary for each trainee. Similar to the personal attendance record. Can you share the doc on here?

**Comment by JoannaRumbley** [comment]
That makes sense. Let's go with that. 😄 

...but the doc only says that one sentence about it. The rest of it is on the other Class Notes pages. (Br Andrew said the other pages may not be necessary, but that they needed the Detail page.) Here's the doc anyway:
https://www.dropbox.com/search/personal?path=%2F&preview=Class+Notes.docx&query=class+notes&search_session_id=27804694195025783996864711989360&search_token=AP0SjwJ969LUwMd0fj%2BAguyNrMvmiRvOxxs9Ra32qV0%3D

And here's a pic of what the doc says about Detail (just for your convenience):
image

**Comment by iamsitting** [comment]
@samvictor232 Try to review this PR. First check that all the desired functionalities are met. Then try check the code quality (redundant code, hard-coded lines, etc.). Lastly, note any UI items that can be improved.

File: ap/classnotes/templates/classnotes/_trainee_list.html [hunk 1]
```diff
@@ -0,0 +1,111 @@
+{% load common_tags %}
+ <h1> Class Notes</h1>
+  {% if classnotes %}
```
Comment (samvictor232) [inline ap/classnotes/templates/classnotes/_trainee_list.html]: If statement not working

**Comment by samvictor232** [comment]
@iamsitting 
Trainee view:
Save button doesn't save, instead it submits the classnote. When the classnote is pending, cannot edit it.

File: ap/classnotes/templates/classnotes/assign_classnotes.html [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+  {% for y in week_range %}
+      <option
+       value="{{ y }}" {% if y == weekinit %} selected="selected" {% endif %}
+      >{{ y }}</option>
+  {% endfor %}
+  </select>
   <input id="assign_classnotes" name="assign_classnotes" class="btn btn-success" type="submit" value="Assign"></input>
 </form>
 </div>
+<img src="{% static 'classnotes/img/temp.gif' %}" id="loading" style="display:none"/ >
```
Comment (samvictor232) [inline ap/classnotes/templates/classnotes/assign_classnotes.html]: Change to match lifestudy loading gif

**Comment by samvictor232** [comment]
@iamsitting 
When in /approve-classnotes, the save button works but it takes you to a blank page.

**Comment by iamsitting** [comment]
@samvictor232 addressed your comments.

**Comment by samvictor232** [comment]
@iamsitting 
Trainee cannot check and save the "Submitting paper copy" checkbox. Classnotes_detail
Also, the submit button is disabled for the trainees. 

On TA side:
When you edit a classnote and check the "submitting paper copy" checkbox and try to save, results in ValidationError

**Comment by iamsitting** [comment]
@samvictor232 comments addressed. Trainees need a minimum word count to submit.

**Comment by iamsitting** [comment]
@samvictor232 try again.

**Comment by samvictor232** [comment]
It works

**Comment by JoannaRumbley** [comment]
Mostly works, from what I can see on the live server:

- [ ] Reports => Class Notes Detail page. The number Owed is opposite. It shows "0" for the ones not yet submitted and "1" owed for the ones already submitted.

**Pull request merged**